### ISSUE TITLE
Discovery of alertmanagers is undocumented

### ISSUE THREAD

**Issue by ds0nt** [coreos/prometheus-operator#322]
**What did you do?**
Try and link a prometheus with alertmanagers, but having trouble. It's not mentioned in the documentation that it discovers via regex on the service's name.

https://github.com/coreos/prometheus-operator/blob/88f90dc68f2e47e2b7ff7ad753396e891b23f8dd/pkg/prometheus/promcfg.go#L300

```yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: Prometheus
metadata:
  name: example
spec:
  replicas: 2
  alerting:
    alertmanagers: # this bit isn't described in the docs
    - namespace: default
      name: alertmanager-example
      port: web
  serviceMonitorSelector:
    matchLabels:
      team: frontend
  resources:
    requests:
      memory: 400Mi
  ruleSelector:
    matchLabels:
      role: prometheus-rulefiles
      prometheus: example
```

**Comment by fabxc** [comment]
That the configuration in your TPR maps to a regex in the generated Prometheus configuration is an implementation detail.  
Generally, most of the generated configuration are a bunch of regexes. We therefore document the abstraction our TPR definitions expose. 

Did you figure out what the problem was? And how could we adapt the documentation to assist in this in the future? I think in general educating people on all the complex configuration generation defeats the purpose of providing an easier way to structure monitoring setups through the operator.

I see that there might be a misalignment of configuring by referencing an endpoints object but than using relabeling on a the service name. This should not matter in general as every endpoints object should map 1:1 to a service.  
But we could switch to relabeling on `__meta_kubernetes_endpoints_name` to be on the safe side.

**Comment by ds0nt** [comment]
Perhaps if it said something like: "The name field of an alertmanager must match the name of the service above". It just doesn't let me know how the mapping happens at all. No mention of the 'name' needing to point to the services name. 

This was important to me as I didn't leave the names as their defaults in the yamls.

![2017-04-29-173920_693x471_scrot](https://cloud.githubusercontent.com/assets/1015732/25559105/cecc3648-2d02-11e7-9fc2-f998788a1008.png)

**Comment by ds0nt** [comment]
perhaps right after the code snippet in the screenie

**Comment by brancz** [comment]
While I agree we could point this out better in the described documentation, it actually doesn't have to be the name of a `Service`, but rather an `Endpoints` object. Whenever you create a `Service` object in Kubernetes a backing `Endpoints` object is created, where the `Service` writes the `Pod`s selected by it into. The `Endpoints` object is what is actually used to discover the Alertmanager. As @fabxc mentioned that this results in a regex in the configuration is merely an implementation detail, and is not important to the user in this case (this may even change in the future without the behavior changing for the user).

@ds0nt would you like to create a PR to add this to the documentation?

**Comment by ds0nt** [comment]
Yep, I'll look into it this evening. (in EST)

**Comment by brancz** [comment]
@ds0nt do you want us to pick up on this or do you think you can do a PR for this?

**Comment by ds0nt** [comment]
Yeah, sorry about that. 

I'm not sure I know what to write, so it might be faster if you guys picked up on it.

**Comment by ds0nt** [comment]
How much would I write about the endpoints object? 

Or could I just say something simple like "make this field here match that field there".

If something simple like that will do, then I can write it ;)

**Comment by brancz** [comment]
I think a simple like "create a service and put the name there" in a few sentences suffices. Thanks!

**Comment by ds0nt** [comment]
https://github.com/coreos/prometheus-operator/pull/379

**Comment by brancz** [comment]
Closed by #379
